### PR TITLE
Upgraded to go 1.17

### LINKED ISSUE #172: Upgrade to go  1.17
go 1.17 was released https://go.dev/blog/go1.17.

The project should upgrade to v1.17

### PR DESCRIPTION
**Pull request by neil465**
Upgraded to go 1.17.

fixes #172

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -55,3 +55,75 @@ require (
 	github.com/mitchellh/mapstructure v1.1.2
 	github.com/spf13/cast v1.4.0
 )
+
+require (
```
Comment (roneli) [inline go.mod:59]: why did we double the require, can you run go mod tidy ?

**Review by roneli — CHANGES_REQUESTED** [review]
Thanks you for your contribution, I have just one minor comment before I can run tests and merge.

File: go.mod (hunk 1 above)
Comment (neil465) [reply go.mod:59]: I already ran go mod tidy and that is the result of this. Thanks

**Comment by roneli** [comment]
Created a new PR without required #178

**Pull request closed without merging**